### PR TITLE
Run nginx without root, without using `nginxinc/nginx-unprivileged`

### LINKED ISSUE #1141: update nginx image used
use original nginx image and manually set a non root user in the dockerfile, instead of using the nginx-unprivileged image

### PR DESCRIPTION
**Pull request by netr0m**
## What does this pull request change?
- Separates `nginx` config files into separate subdirectories
- Uses a [shared file for proxy headers](https://github.com/equinor/data-modelling-tool/blob/356bb4b8da162756b943deb5a77410bfcb1b1795/nginx/config/proxy.conf)
- Runs the Docker container as user `nginx`
- Specifies `user nginx` in the nginx config
- Uses the default `nginx` Docker image, instead of relying on [nginx-unprivileged](https://hub.docker.com/r/nginxinc/nginx-unprivileged)

## Why is this pull request needed?
- To avoid relying on third party Docker image, which among other things may lag behind on updates/upgrades
- To allow for rootless Docker

## Issues related to this change
- Resolves #1141

### CODE REVIEW

File: nginx/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+FROM nginx:latest
+
 COPY nginx.conf /etc/nginx/nginx.conf
-COPY security.conf /etc/nginx/security.conf
-COPY default.nginx /etc/nginx/conf.d/default.conf
-USER 1000
\ No newline at end of file
+
+# configs
+COPY config/general.conf /etc/nginx/config/general.conf
```
Comment (soofstad) [inline nginx/Dockerfile:6]: Can this be done in one copy command? Or is there files you dont want to override

File: nginx/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,21 @@
-FROM nginxinc/nginx-unprivileged:1.20
+FROM nginx:latest
```
Comment (soofstad) [inline nginx/Dockerfile:1]: Can we use alpine tag here?

File: nginx/sites-available/default.conf [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    location / {
+        proxy_pass          http://web:3000/;
+
+        include             /etc/nginx/config/general.conf;
+        include             /etc/nginx/config/proxy.conf;
+        include             /etc/nginx/config/websocket.conf;
+    }
+}
+
+# server {
```
Comment (soofstad) [inline nginx/sites-available/default.conf:44]: Remove this?

**Review by soofstad — CHANGES_REQUESTED** [review]

File: nginx/Dockerfile (hunk 2 above)
Comment (netr0m) [reply nginx/Dockerfile]: Yeah, good idea. Switched now

**Review by soofstad — APPROVED** [review]

**Pull request merged**